scripts/bootloader/generate_zip.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+import argparse
+import json
+import time
+from zipfile import ZipFile
+from os import path, remove
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Generate zip file for binary artifact. Provide a list of 'key=value' pairs separated by space for"
+                    "information that should be stored in manifest.json. To provide information specific for one of "
+                    "the files, prepend the 'key' with the name of the file (only the basename, not the path)",
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+
+    parser.add_argument("--output", required=True, type=argparse.FileType(mode='w'), help="Output zip path")
+    parser.add_argument("--bin-files", required=True, type=argparse.FileType(mode='r'), nargs='+',
+                        help="Bin files to be stored in zip")
+    return parser.parse_known_args()
+
+
+def get_name(text_wrapper):
+    return path.basename(text_wrapper.name)
+
+
+if __name__ == "__main__":
+    args, info = parse_args()
+
+    manifest = dict()
+
+    if path.exists(args.output.name):
+        remove(args.output.name)
```

Comment:
This should be unnecessary.

opening a file in 'w' mode will "remove" it first.

If there is something exceptional going on then this should be commented, but I hope we can get rid of this code.

Reply:
Good point! Removing this (and making a note to future self)